### PR TITLE
Digeq 115 author validation rules

### PR DESCRIPTION
**Pull request by warren-methods**
**What**
Allows a user to author validation rules for a particular question. All questions have the "required" validation rule as default. Text question also have maximum length. Numeric questions have greater than, less than, equal to and not equal to.

**How to test**
1. Check out this branch and run the author application
2. Create a new questionnaire
3. Add the various question types, checking you can add the validation rules as required by the story.

**Who can review**
Anyone apart from @warren-methods

### CODE REVIEW

**Comment by dhilton** [comment]
When you toggle the preview - the tabs still display which is from the builder author view, think it's just a question of display noning on that event.

**Comment by warren-methods** [comment]
Good spot, will change it

File: author/static/js/q_builder/templates/text_question.html [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+						<li ng-repeat="condition in child.validation">
+								<div ng-if="condition.condition == 'required'">
+									<label>Required</label>
+									<input type="checkbox" ng-model="condition.value" ng-checked="condition.value"/>
+								</div>
+								<div ng-if="condition.condition == 'maxlength'">
+									<label>Please enter the maximum number of characters the user can enter</label>
+									<input dnd-nodrag type="text" ng-model="condition.value" placeholder="E.g - 500 ">
+									<label>Validation Type</label>
+									<select value="errortype" ng-model="condition.type">
```
Comment (dhilton) [inline author/static/js/q_builder/templates/text_question.html]: After discussions with Ben, can we change the copy on the error types to "Error - (Hard check)" and "Warning - (Soft check)" as hard and soft are well known semantics for what we mean by these types of errors.

**Comment by dhilton** [comment]
After conversation with everyone, we're going to remove the standard error text box and move the editing of standard required error checks to the validation tab.

**Comment by dhilton** [comment]
:+1:  This looks good to merge BUT let's wait until the Survey runner side is ready as this is a breaking change for our system - merging the two at the same / similar time.

**Comment by warren-methods** [comment]
Rebased

**Pull request merged**